lib/permutat.g

```diff
@@ -479,6 +479,64 @@ BIND_GLOBAL( "ListPerm", function( arg )
 end );
 
 
+#############################################################################
+##
+#F  CycleFromList( <list> )  . . . . . . . . . . . cycle defined from a list 
+##
+##  <#GAPDoc Label="CycleFromList">
+##  <ManSection>
+##  <Func Name="CycleFromList" Arg='list'/>
+##
+##  <Description>
+##  For the given dense, duplicate-free list of positive integers
+##  <M>[a_1, a_2, ..., a_n]</M>
+##  return the <M>n</M>-cycle <M>(a_1,a_2,...,a_n)</M>. For the empty list
+##  the trivial permutation <M>()</M> is returned.
+##  <P/>
+##  If the given <A>list</A> contains duplicates or holes, return <K>fail</K>.
+##  <P/>
+##  <Example><![CDATA[
+##  gap> CycleFromList( [1,2,3,4] );
+##  (1,2,3,4)
+##  gap> CycleFromList( [3,2,6,4,5] );
+##  (2,6,4,5,3)
+##  gap> CycleFromList( [2,3,2] );
+##  fail
+##  gap> CycleFromList( [1,,3] );
+##  fail
+##  ]]></Example>
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+BIND_GLOBAL( "CycleFromList", function( list )
+    local max, images, set, i, min;
+
+    # Trivial case
+    if Length(list) = 0 then
+        return ();
+    fi;
+
+    set := Set(list);
+    if Length(set) <> Length(list) then
+        # we found duplicates (or list was not dense)
+        return fail;
+    fi;
+    min := Minimum( set );
+    if min <= 0 then
+        Error("CycleFromList: List must contain only positive integers.");
+    fi;
+    max := Maximum( set );
+    images := [1..max];
+    for i in [1..Length(list)-1] do
+        images[ list[i] ] := list[i+1];
+    od;
+    images[ list[Length(list)] ] := list[1];
+
+    return PermList(images);
```

Comment:
While manual test cases are provided, and exercise the regular parts of the function, not all failure cases are tested, e.g. `CycleFromList([0])`, `CycleFromList([1/2])`, `CycleFromList([true])` to name a few. Perhaps add some (though perhaps to a .tst file, not the manual example)?

I'll not block the PR over this, but it is needed to maximize the test coverage.

Reply:
There is also no test for the empty list. All this can be seen on <https://codecov.io/gh/gap-system/gap/pull/2242/diff?src=pr&el=tree#diff-bGliL3Blcm11dGF0Lmc=> (link comes from the codecov report in the PR)

Reply:
I don't know much about the test directory structure of GAP. Where exactly should I put the tests?

Reply:
Somewhere into `tst/testinstall/`. In this case, I'd suggest to create `tst/testinstall/opers/CyclesFromList.tst`.  Look at the other files in that directory to get an idea what to put in, in case you are not familiar with .tst files. And feel free to ask more questions.